### ISSUE TITLE
Cryptic error when CSV file heading misnamed

### ISSUE THREAD

**Issue by willu47** [rgieseke/pandas-datapackage-reader#14]
The packages returns the cryptic error `None of [expect column name] are in the columns` when a heading line in one of CSV files contained in the data package contains an incorrect column name.

It would be more useful if the error returned the name of the resource in which this error is raised.

In the traceback below, I'm calling the pandas-datapackage-reader from my `narrow_to_datafile` module.

```python
Traceback (most recent call last):
  File "/Users/wusher/miniconda3/envs/otoole/bin/otoole", line 11, in <module>
    load_entry_point('otoole', 'console_scripts', 'otoole')()
  File "/Users/wusher/repository/otoole/src/otoole/cli.py", line 307, in main
    args.func(args)
  File "/Users/wusher/repository/otoole/src/otoole/cli.py", line 138, in conversion_matrix
    convert_datapackage_to_datafile(args.from_path, args.to_path)
  File "/Users/wusher/repository/otoole/src/otoole/preprocess/narrow_to_datafile.py", line 225, in convert_datapackage_to_datafile
    dp = DataPackageToCsv(path_to_datapackage, path_to_datafile)
  File "/Users/wusher/repository/otoole/src/otoole/preprocess/narrow_to_datafile.py", line 34, in __init__
    self.package = self._get_package()
  File "/Users/wusher/repository/otoole/src/otoole/preprocess/narrow_to_datafile.py", line 44, in _get_package
    package = read_datapackage(self.datapackage)  # typing: datapackage.Package
  File "/Users/wusher/miniconda3/envs/otoole/lib/python3.7/site-packages/pandas_datapackage_reader/__init__.py", line 147, in read_datapackage
    df = df.set_index(index_col)
  File "/Users/wusher/miniconda3/envs/otoole/lib/python3.7/site-packages/pandas/core/frame.py", line 4303, in set_index
    raise KeyError(f"None of {missing} are in the columns")
KeyError: "None of ['FUEL'] are in the columns"
```

**Comment by rgieseke** [comment]
Is the error always happening with a wrong column name or only when the `primaryIndex` is wrong?
The traceback errors with the index setting in `pandas-datapackage-reader`.
https://github.com/rgieseke/pandas-datapackage-reader/blob/master/pandas_datapackage_reader/__init__.py#L147

**Comment by willu47** [comment]
Hi @rgieseke - You're right I believe it is related to the missing/wrong primary index.  Still, it would be useful to get a report of the resource which triggers the error.

**Comment by rgieseke** [comment]
Definitely! Should be no problem to wrap all reading inside the for loop in a try/except, print the resource name on error and then the original traceback/error message.

Would you be interested in submitting a PR for that?

**Comment by rgieseke** [comment]
Fixed in #15
